server/compute.js

```diff
@@ -143,6 +143,11 @@ module.exports = function compute({ changes = {}, user, segments, events = [] },
     if (d.kind === "N" || d.kind === "E") {
       _.set(memo, d.path, d.rhs);
     }
+    // when we have an array updated we set the whole
+    // array in `changed` constant
+    if (d.kind === "A") {
+      _.set(memo, d.path, _.get(updatedUser, d.path, []));
```

Comment:
Yes, I'm seeing that is a deeper case, I also thought about tests here, really scary to modify it without testing.